inbm/dispatcher-agent/dispatcher/sota/downloader.py

```diff
@@ -165,3 +168,66 @@ def download(self,
 
     def check_release_date(self, release_date: Optional[str]) -> bool:
         return self.is_valid_release_date(release_date)
+
+
+class TiberOSDownloader(Downloader):
+    """TiberOSDownloader class, child of Downloader
+
+       @param signature: signature used to preform signature check on downloaded image.
+    """
+
+    def __init__(self, signature: Optional[str] = None) -> None:
+        super().__init__()
+        self._signature = signature
+
+    def download(self,
+                 dispatcher_broker: DispatcherBroker,
+                 uri: Optional[CanonicalUri],
+                 repo: IRepo,
+                 username: Optional[str],
+                 password: Optional[str],
+                 release_date: Optional[str]) -> None:
+        """Downloads files and places image in local cache
+
+        @param dispatcher_broker: DispatcherBroker object used to communicate with other INBM services
+        @param uri: URI of the source location
+        @param repo: repository for holding the download
+        @param username: username to use for download
+        @param password: password to use for download
+        @param release_date: manifest release date
+        @raises SotaError: release date is not valid
+        """
+
+        if uri is None:
+            raise SotaError("URI is None while performing TiberOS download")
+
+        if password is None:
+            raise SotaError("JWT Token is None while performing TiberOS download")
+
+        oras_download(dispatcher_broker=dispatcher_broker,
+                      uri=uri,
+                      repo=repo,
+                      umask=UMASK_OTA,
+                      username=username,
+                      password=password)
+
+        # Perform signature check.
+        # Multiple files may have been downloaded from OCI.
+        # The method below will iterate over all files in the repo, calculate the SHA256sum for each file,
+        # and compare it with the provided signature.
+        if self._signature:
+            logger.debug("Perform signature check on the downloaded file.")
+            dest_repo = repo.get_repo_path()
+            for filename in os.listdir(dest_repo):
```

Comment:
a lot of these `os` calls could raise exceptions

Reply:
Added try/except to catch OSError.